cmd/api/src/api/v2/agi.go

```diff
@@ -258,7 +259,29 @@ func (s Resources) UpdateAssetGroupSelectors(response http.ResponseWriter, reque
 		if result, err := s.DB.UpdateAssetGroupSelectors(*ctx.FromRequest(request), assetGroup, selectorSpecs, false); err != nil {
 			api.HandleDatabaseError(request, response, err)
 		} else {
-			// When T0 asset group selectors are modified we must trigger analysis
+			// Update system tags for the updated selectors
+			nodeProperties := graph.NewProperties()
+			if assetGroup.Tag == model.TierZeroAssetGroupTag {
+				nodeProperties.Set(common.SystemTags.String(), model.TierZeroAssetGroupTag)
+			} else {
+				nodeProperties.Set(common.SystemTags.String(), model.OwnedAssetGroupTag)
+			}
```

Comment:
I'm not sure if you are soliciting feedback on this MR yet however I think we should simplify this logic to just apply or remove `assetGroup.Tag` based on if the selector is being created or deleted. There may be a future where we introduce additional asset groups in which case defaulting to `model.OwnedAssetGroupTag` is not the desired behavior.